sklearn/metrics/scorer.py

```diff
@@ -514,6 +514,13 @@ def make_scorer(score_func, greater_is_better=True, needs_proba=False,
 log_loss_scorer = make_scorer(log_loss, greater_is_better=False,
                               needs_proba=True)
 log_loss_scorer._deprecation_msg = deprecation_msg
+# Currently brier_score_loss don't support the shape of result
```

Comment:
But surely we need the same for `roc_auc_score` (although that can fall back on `decision_function` when available). What's going on in that case?